collector/processor/concurrentbatchprocessor/batch_processor.go

```diff
@@ -565,7 +565,12 @@ func (bt *batchTraces) sizeBytes(data any) int {
 	return bt.sizer.TracesSize(data.(ptrace.Traces))
 }
 
-func (bt *batchTraces) export(ctx context.Context, req any) error {
+func (bt *batchTraces) export(ctx context.Context, req any) (retErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			retErr = errors.New(r.(string))
+		}
+	}()
```

Comment:
I'd like to see a new top-level function--
```
func recoverError(retErr *error) {
		if r := recover(); r != nil {
			*retErr = errors.New(fmt.Sprint(r))
		}
	}
```

Also note, the `r` value will be whatever is passed to `panic`, so not always a string. Suggest letting `fmt.Sprint` figure out the best to-string method.

Reply:
(So, `fmt.Errorf("%v", r)` also works, no need for `errors.New()` then.)

Reply:
Thanks for the review! Fixed